testsuite/tests/lib-list/test.ml

```diff
@@ -42,6 +42,7 @@ let () =
   assert (List.compare_length_with ['1'] 1 = 0);
   assert (List.compare_length_with ['1'] 2 < 0);
   assert (List.filter_map string_of_even_opt l = ["0";"2";"4";"6";"8"]);
+  assert (List.fold_map (fun a b -> a + b, b) 0 l = (45, l));
```

Comment:
nit, but this test case doesn't check that the `map` part of `fold_map` does anything.

Reply:
I added some more tests.